### DISCUSSION TITLE
Blob from file (iOS and android)

### DISCUSSION THREAD

**Discussion by babycourageous** [capawesome-team/capacitor-file-picker#85] [Q&A] [answered]
hey there!

Wondering if anyone has advice on using the FilePicker (Ionic/Capacitor project) to pick a video file and then convert to a Blob for an upload using FormData. Tried converting to Base64 but larger files seem to crash so that's a bust.

Thanks!

**Comment by robingenz** [comment, marked as answer]
You need to read the picked file at the provided path using the [Capacitor Filesystem](https://capacitorjs.com/docs/apis/filesystem) plugin.
But, as you said, this can lead to a crash.

This is what I posted about the topic the other day on the official Ionic Discord server:

> IMHO the perfect solution is to handle files completely native (upload/download/open/etc.) on Android and iOS. No Base64 etc. Capacitor gives you everything you need with [`convertFileSrc(...)`](https://capacitorjs.com/docs/core-apis/web#convertfilesrc) and the possibility to write native code in the form of plugins. For example, I created a Capacitor Cloudinary plugin (https://github.com/capawesome-team/capacitor-cloudinary) for a client to upload and download files directly without going through the webview. We migrated his app to native file handling only. The performance and UX improvement was great. I am also planning to publish a blog post about this.

**Comment by babycourageous** [comment]
Oh wow - i was just reading that thread! haha
Would you say looking at the code for Capawesome Cloudinary is a good start to wrap my head around native file handling?

Thanks for the info!

**Reply by robingenz** [reply]
Unfortunately not, because I used the Cloudinary SDK for most parts. But should be able to use https://github.com/capacitor-community/http to upload files.

**Comment by babycourageous** [comment]
Thanks for the convo @robingenz !

**Reply by jayroyall** [reply]
Hi, did you ever manage to upload the video once it had been selected? I'm currently facing the same issue so any pointers would be greatly appreciated!

Thanks,

Jay

**Reply by babycourageous** [reply]
Hi @jayroyall - I haven't actually had time to look into the abstractions to handle files natively as Robin suggested. My needs related to cloudinary so for my use - i just opted to use the Capawesome Cloudinary plugin rather than sending the flie to the server to upload to Cloudinary. 
If I do get around to trying to write an abstraction I'll happily post the results here!

**Reply by jayroyall** [reply]
OK no probs, thanks for getting back to me. If I find a solution to my problem then I'll let you know.

Thanks again, Jay
